src/Modules/OrchardCore.Commerce.Payment/Services/PaymentService.cs

```diff
@@ -198,6 +198,21 @@ await _paymentProvidersLazy
             orderPart.LineItems.SetItems(lineItems);
             orderPart.Status.Text = OrderStatuses.Pending.HtmlClassify();
 
+            // When billing and shipping addresses should match, if an address field is null but the other one is not,
+            // fill out the null field's data.
+            if (orderPart.BillingAndShippingAddressesMatch.Value)
+            {
+                if (orderPart.BillingAddress.Address.Name is null && orderPart.ShippingAddress.Address.Name is not null)
+                {
+                    orderPart.BillingAddress = orderPart.ShippingAddress;
+                }
+
+                if (orderPart.ShippingAddress.Address.Name is null && orderPart.BillingAddress.Address.Name is not null)
+                {
+                    orderPart.ShippingAddress = orderPart.BillingAddress;
+                }
```

Comment:
```suggestion
                billingAddress ??= shippingAddress;
		shippingAddress ??= billingAddress;
```

Reply:
The `BillingAddress` and `ShippingAddress` properties [are never null](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/18a900038099da680ea49c99c4ab63732564235f/src/Libraries/OrchardCore.Commerce.Abstractions/Models/OrderPart.cs#L42-L43). Also their `Address` properties (e.g. `OrderPart.BillingAddress.Address`) [are always initialized as well](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/18a900038099da680ea49c99c4ab63732564235f/src/Libraries/OrchardCore.Commerce.Abstractions/Fields/AddressField.cs#L8). That's why it's necessary to drill down and individually check the name field (which is a known required field) instead of your suggestion.

@porgabi On the other hand I don't `not null` checks of each condition are necessary. It only matters if both addresses are empty, but then you would get the same exception as before anyway. So it's cleaner and probably slightly faster to remove them.